jenkins/deploy-clio-server.sh

```diff
@@ -170,59 +154,126 @@ render_ctmpls() {
   rm ${ctmpl_env_file}
 }
 
-stop_containers() {
-  local -r clio_fqdn=$1
+activate_service_account() {
+  VAULT_TOKEN=$(cat /etc/vault-token-dsde)
+  SA_VAULT_PATH="secret/dsde/gotc/dev/common/ci-deployer-service-account.json"
+
+  docker run --rm -e VAULT_TOKEN="${VAULT_TOKEN}" \
+    broadinstitute/dsde-toolbox:dev vault read -format=json ${SA_VAULT_PATH} \
+    | jq '.data' > service-account.json
 
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "test -f ${COMPOSE_FILE} && ${DOCKER_COMPOSE[*]} stop || echo"
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "test -f ${COMPOSE_FILE} && ${DOCKER_COMPOSE[*]} rm -f || echo"
+  # activate SA
+  gcloud auth activate-service-account --key-file=service-account.json
 }
 
-start_containers() {
-  local -r clio_fqdn=$1
+get_clio_instance() {
+  CLIO_PROJECT=$1
 
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "${DOCKER_COMPOSE[*]} pull"
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "${DOCKER_COMPOSE[*]} up -d"
+  # Get the instance marked with the Clio and Active labels (There should only be one)
+  CLIO_INSTANCE_LIST=$(gcloud --format="table[no-heading](Name)" compute \
+                         --project ${CLIO_PROJECT} instances list \
+                         --filter="labels.app=clio AND labels.state=active")
+
+  # Verify that there is only one instance
+  if [[ ${#CLIO_INSTANCE_LIST[@]} -ne 1 ]];
+  then
+    exit 1
+  fi
+
+  echo "${CLIO_INSTANCE_LIST}"
 }
 
-deploy_containers() {
-  local -r clio_fqdn=$1 config_src=$2
+get_clio_fqdn() {
+  CLIO_PROJECT=$1
+  CLIO_INSTANCE=$2
+
+  local -r gce_name=$(gcloud compute --project ${CLIO_PROJECT} \
+                                     ssh ${CLIO_INSTANCE} \
+                                     --zone=us-central1-a \
+                                     --command="uname -n")
+  local -r clio_cluster_number_long=$(echo "${gce_name}" | sed -E "s/clio-([0-9]{3})-([0-9]{2})/\1/")
+  local -r clio_cluster_number=${clio_cluster_number_long:0:1}
+  local -r instance_number=$(echo "${gce_name}" | sed -E "s/clio-([0-9]{3})-([0-9]{2})/\2/")
+  local -r clio_fqdn=$(build_fqdn "clio${clio_cluster_number}${instance_number}")
+  echo "${clio_fqdn}"
+}
+
+stop_clio_containers() {
+  CLIO_PROJECT=$1
+  CLIO_INSTANCE=$2
+
+  gcloud compute --project ${CLIO_PROJECT} \
+                 ssh ${CLIO_INSTANCE} \
+                 --zone=us-central1-a \
+                 --command="test -f ${COMPOSE_FILE} &&
+                              ${DOCKER_COMPOSE[*]} stop || echo &&
+                            test -f ${COMPOSE_FILE} &&
```

Comment:
Why is the second `test` here? If `COMPOSE_FILE` wasn't there when the first `test` runs, then this line wouldn't be executed. Or is the thinking that maybe `docker-compose stop` might delete the compose.yml file, so we need to check after running `stop`?

Reply:
This was a copy-paste error. It was originally 2 different commands and they got merged into one. I removed the second test.